inc/azure_c_util/interlocked_if.h

```diff
@@ -0,0 +1,156 @@
+#ifndef INTERLOCKEDIFIF_H
```

Comment:
We should have the copyright comment for all header and source files as the first line:

// Copyright (C) Microsoft Corporation. All rights reserved.